### PR TITLE
feat(MGDCTRS-639): makes operators cluster-wide

### PR DESCRIPTION
**Pull request by rinaldodev**
Changes the behavior of debezium and camel operators so they will watch
the entire cluster for changes on ManagedConnector and other resources.

The property cos.connectors.namespace is not used anymore. Resources are
now created in the same namespace where the ManagedConnector resource
already is.

Roles and RoleBindings were changed to cluster.

### CODE REVIEW

**Review by lburgazzoli — CHANGES_REQUESTED** [review]
The roles & role binding need to be updated to be cluster scoped.

i.e. in https://github.com/rinaldodev/cos-fleetshard-1/blob/feat/cluster-mode-operators/cos-fleetshard-operator-camel/src/main/kubernetes/kubernetes.yml, we define a `cos-fleetshard-operator-camel` role that lists the resources the operator can access using `rbac.authorization.k8s.io/v1:Role` however, that's limits the visibility to the namespace in which the operator runs.

We should migrate:
- Role -> ClusterRole
- RoleBinding -> ClusterRoleBinding

File: cos-fleetshard-it/cucumber/src/main/java/org/bf2/cos/fleetshard/it/cucumber/ConnectorContext.java [hunk 1]
```diff
@@ -52,7 +51,7 @@
     private final List<ManagedConnector> history = new CopyOnWriteArrayList<>();
 
     @Inject
-    private CosFeatureContext cosCtx;
+    CosFeatureContext cosCtx;
```
Comment (valdar) [inline cos-fleetshard-it/cucumber/src/main/java/org/bf2/cos/fleetshard/it/cucumber/ConnectorContext.java:54]: Why the change in scope?

**Review by valdar — CHANGES_REQUESTED** [review]

File: cos-fleetshard-it/cucumber/src/main/java/org/bf2/cos/fleetshard/it/cucumber/ConnectorContext.java (hunk 1 above)
Comment (rinaldodev) [reply cos-fleetshard-it/cucumber/src/main/java/org/bf2/cos/fleetshard/it/cucumber/ConnectorContext.java:54]: Quarkus complains since it has to use reflection to inject it, and from looking at other Injection points we seem to be following this pattern of package-private.

**Review by valdar — APPROVED** [review]

**Comment by heiko-braun** [comment]
@rinaldodev I guess this requies a larger change to the managed-tenants addon as well. Do you plan to provide it too?

**Review by lburgazzoli — APPROVED** [review]
